xarray/core/dataset.py

```diff
@@ -851,6 +871,9 @@ def selkeys(dict_, keys):
             return dict((d, dict_[d]) for d in keys if d in dict_)
 
         def maybe_chunk(name, var, chunks):
+            if name not in self.data_vars:
```

Comment:
I see your point about performance, but I think that mostly holds true for indexes. So I would be inclined to adjust this to only skip variables in `self.dims` (aka indexes used for alignment).

I am still concerned about skipping coords if they are already dask arrays. If they are already dask arrays, then `.chunk()` should probably adjust their chunks anyways.